[Swift in WebKit] Stop accidentally exposing WebKitSwift as SPI

rr-codes opened the pull request:
#### cb3d85a1036043254c10ac4e2299d639c4a1843b
<pre>
[Swift in WebKit] Stop accidentally exposing WebKitSwift as SPI
<a href="https://bugs.webkit.org/show_bug.cgi?id=292361">https://bugs.webkit.org/show_bug.cgi?id=292361</a>
<a href="https://rdar.apple.com/150424287">rdar://150424287</a>

Reviewed by Elliott Williams.

WebKitSwift is currently somewhat exposed as SPI due to various headers mistakenly using &quot;private&quot; visability.
Correct this, and fix other inconsistent WKS infrastructure issues.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Enforce consistency with regards to header visibility across all WKS headers. Specifically, each
header should be part of both WebKit and WebKitSwift, and should have project-level visibility for each.

* Source/WebKit/WebKitSwift/TextExtraction/WKTextExtractionItem.h: Renamed from Source/WebKit/UIProcess/Cocoa/WKTextExtractionItem.h.

Enforce consistency with regards to the location of WKS types. All headers for WKS types should be in WKS.

* Source/WebKit/WebKitSwift/WebKitSwift.h:

Make this header project-level. In addition, since each child header file is also project, use normal quotes
instead of angle quotes for the imports.

* Source/WebKit/WebKitSwift/module.modulemap:

The header need not be umbrella, so remove this. This makes it more consistent with WebKitInternal.h

* Tools/WebKitTestRunner/cocoa/WKTextExtractionTestingHelpers.mm:

This file is a weird edge-case wherein a different project needs access to a WKS type. For this case particularly,
this is resolved by forward-declaring the types. Text extraction will eventually move to WK proper, so this is
just temporary anyways.

Canonical link: <a href="https://commits.webkit.org/294413@main">https://commits.webkit.org/294413@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/180b4b01dc158d87cde4a7b9ccf6435c2806d8c6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101752 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21420 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11736 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106910 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/52386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/103792 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21728 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29922 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/77470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/52386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104759 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16787 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91883 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57808 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16612 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9901 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51737 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/86470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9978 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109266 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28885 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/21271 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/86451 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29246 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88084 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86019 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21887 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30777 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/23048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28813 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34103 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28624 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31947 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30183 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Tools/WebKitTestRunner/cocoa/WKTextExtractionTestingHelpers.mm:
```diff
@@ -26,14 +26,64 @@
 #import <wtf/text/WTFString.h>
 #import <wtf/text/cf/StringConcatenateCF.h>
 
+// FIXME: Remove these forward declarations once text extraction support is fully migrated to WebKit from WebKitSwift.
```
How does migrating to WebKit make it easier to access these? Would the header need to be made WebKit SPI?

rr-codes replied:
the header was in WebKit as project-level originally, and this file was able to import it (I have no idea how, but it worked). so putting it back in WK would restore this to the way it was before

emw-apple reviewed (approved):
Seems reasonable, and a step towards making it easier to move some of this code into the main WebKit dylib.

rr-codes committed:
[Swift in WebKit] Stop accidentally exposing WebKitSwift as SPI
https://bugs.webkit.org/show_bug.cgi?id=292361
rdar://150424287

Reviewed by Elliott Williams.

WebKitSwift is currently somewhat exposed as SPI due to various headers mistakenly using "private" visability.
Correct this, and fix other inconsistent WKS infrastructure issues.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Enforce consistency with regards to header visibility across all WKS headers. Specifically, each
header should be part of both WebKit and WebKitSwift, and should have project-level visibility for each.

* Source/WebKit/WebKitSwift/TextExtraction/WKTextExtractionItem.h: Renamed from Source/WebKit/UIProcess/Cocoa/WKTextExtractionItem.h.

Enforce consistency with regards to the location of WKS types. All headers for WKS types should be in WKS.

* Source/WebKit/WebKitSwift/WebKitSwift.h:

Make this header project-level. In addition, since each child header file is also project, use normal quotes
instead of angle quotes for the imports.

* Source/WebKit/WebKitSwift/module.modulemap:

The header need not be umbrella, so remove this. This makes it more consistent with WebKitInternal.h

* Tools/WebKitTestRunner/cocoa/WKTextExtractionTestingHelpers.mm:

This file is a weird edge-case wherein a different project needs access to a WKS type. For this case particularly,
this is resolved by forward-declaring the types. Text extraction will eventually move to WK proper, so this is
just temporary anyways.

Canonical link: https://commits.webkit.org/294413@main
